packages/common/src/index-api.ts

```diff
@@ -75,10 +75,17 @@ export type RangeValueFilter<T extends string | number> =
 export type ScalarValueFilter<T extends string | number = string | number> = CommonValueFilter<T> &
   RangeValueFilter<T>
 
+type StringCommonValueFilter = ScalarValueFilter<string>
+type NumberCommonValueFilter = ScalarValueFilter<number>
+
 /**
  * Any supported field value filter on an object
  */
-export type AnyValueFilter = BooleanValueFilter | EnumValueFilter | ScalarValueFilter
+export type AnyValueFilter =
```

Comment:
This change isn't needed, `ScalarValueFilter` already works for both `string` and `number`